src/progress/pct/compile.p

```diff
@@ -499,6 +545,25 @@ PROCEDURE compileXref.
 
 END PROCEDURE.
 
+PROCEDURE printErrorsWarningsJson.
+  IF ( outputType EQ 'json' ) THEN DO:
+    DEFINE VARIABLE httProjectErrors    AS HANDLE.
+    DEFINE VARIABLE httProjectWarnings  AS HANDLE.
+    
+    httProjectErrors   = TEMP-TABLE ttProjectErrors:HANDLE.
+    httProjectWarnings = TEMP-TABLE ttProjectWarnings:HANDLE.
+
+    DEFINE VARIABLE ttProjectErrorsFile AS CHARACTER NO-UNDO.
+    ASSIGN ttProjectErrorsFile = PCTDir + '/':U + 'projectErrors.json':U.
+
+    DEFINE VARIABLE ttProjectWarningsFile AS CHARACTER NO-UNDO.
+    ASSIGN ttProjectWarningsFile = PCTDir + '/':U + 'projectWarnings.json':U.
+    
+    httProjectErrors:WRITE-JSON("file", ttProjectErrorsFile).
```

Comment:
Wouldn't it be simpler to generate only one file ?  I'd  also include the number of files being compiled, and the number of failed compilations in the JSON output.

Reply:
The reason I didn't write in a single file due to the behavior of WRITE-JSON.
From Doc:
If a file with the specified name already exists, the AVM verifies that the file is writeable and overwrites the file.
To keep it simple, I am writing in two files as we have two temp tables

Reply:
A dataset will write both TT with a single statement. And I'm sure the number of files will be important in PDSOE.